mapsforge-map/src/main/java/org/mapsforge/map/rendertheme/renderinstruction/Caption.java

```diff
@@ -94,28 +95,6 @@ public Caption(GraphicFactory graphicFactory, DisplayModel displayModel, String
                 this.position = Position.BELOW;
             }
         }
-        switch (this.position) {
-            case CENTER:
-            case BELOW:
-            case ABOVE:
-                this.stroke.setTextAlign(Align.CENTER);
-                this.fill.setTextAlign(Align.CENTER);
-                break;
-            case BELOW_LEFT:
-            case ABOVE_LEFT:
-            case LEFT:
-                this.stroke.setTextAlign(Align.RIGHT);
-                this.fill.setTextAlign(Align.RIGHT);
-                break;
-            case BELOW_RIGHT:
-            case ABOVE_RIGHT:
-            case RIGHT:
-                this.stroke.setTextAlign(Align.LEFT);
-                this.fill.setTextAlign(Align.LEFT);
-                break;
-            default:
-                throw new IllegalArgumentException("Position invalid");
-        }
```

Comment:
Those lines interfere with correct drawing of labels, but only on Android, AWT lives just fine without it:

https://github.com/mapsforge/mapsforge/blob/698cf613f0c0694f0f385440d3ef9e03794f75f7/mapsforge-map-awt/src/main/java/org/mapsforge/map/awt/graphics/AwtPaint.java#L247-L250

So removing label Paint-level align on Android basically makes the code that draws the labels more symmetrical and simpler. Do you think there are any drawbacks to this? Haven't found any.

Reply:
@Sublimis Tests must be made with the `position` parameter of the `caption` rule.
With `master` branch this section is needed, otherwise it breaks the Android rendering.

With the PR code it seems to not be needed. That's why I ask where it is handled now?

Reply:
> @Sublimis Tests must be made with the `position` parameter of the `caption` rule. With `master` branch this section is needed, otherwise it breaks the Android rendering.
> 
> With the PR code it seems to not be needed. That's why I ask where it is handled now?

Here: https://github.com/mapsforge/mapsforge/pull/1579#discussion_r1858330986